pkg/operator/starter.go

```diff
@@ -107,19 +106,6 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
 		cloudConfigBytes,
 		controllerConfig.EventRecorder)
 
-	// targetConfigController is only used for maintaining conditions which
-	// were previously added by the operator
```

Comment:
Why are you removing this?

Reply:
It appears to be causing unncessary churn in conditions added to the CR. I frequently see `TargetConfigControllerDegraded: False` added condition, which only gets removed later on.  So, I removed it. 

Earlier we assumed that, it is tricky to remove conditions from `Status` objects and hence we kept this controller to remove conditions from a controller, we were no longer using. But now I am thinking that, if some cluster has leftover conditions, users can remove it manually.

Reply:
> if some cluster has leftover conditions, users can remove it manually

If that condition is Degraded=true or Available=false created by the old operator, e.g. because API server is not available during upgrade, then upgrade gets stuck.

Reply:
okay, I have brought back the code and made sure it doesn't flip flop anymore.